src/components/LearningMaterialViewer.tsx

```diff
@@ -420,21 +420,16 @@ export default function LearningMaterialViewer({
     // Check for integration blocks and fetch content
     useEffect(() => {
         if (taskData?.blocks && taskData.blocks.length > 0) {
-            const integrationBlock = taskData.blocks.find(block => block.type === 'notion');
+            const integrationBlock = taskData.blocks.find(block => block.type === currentIntegrationType);
             if (integrationBlock) {
                 setIntegrationBlocks(integrationBlock.content);
-                setIntegrationError(null);
-                setIsLoadingIntegration(false);
-            } else {
-                setIntegrationBlocks([]);
-                setIntegrationError(null);
                 setIsLoadingIntegration(false);
+                return;
             }
-        } else {
-            setIntegrationBlocks([]);
-            setIntegrationError(null);
-            setIsLoadingIntegration(false);
         }
+        // Default if no integrationBlock found
+        setIntegrationBlocks([]);
+        setIsLoadingIntegration(false);
```

Comment:
why do we have `setIsLoadingIntegration`? there is no integration being loaded right?